### PR TITLE
docs(cloudflare): update `cloudflare-module`

### PR DESCRIPTION
**Pull request by gxres042**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

With Cloudflare Workers Static Assets, we only need to bind `ASSETS` and don't need to configure `bucket` anymore.Also, a link at Cloudflare Service Workers part has an unnecessary `/`, making the server return 404 to visitor.

This PR resolves these problems.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/2.deploy/20.providers/cloudflare.md [hunk 1]
```diff
@@ -85,16 +85,14 @@ The following shows a typical `wrangler.toml` file for a Nitro application:
 name = "playground"
 main = "./.output/server/index.mjs"
 workers_dev = true
-compatibility_date = "2023-12-01"
+compatibility_date = "2024-11-01"
```
Comment (pi0) [inline docs/2.deploy/20.providers/cloudflare.md:88]: I think we also need to document `compatibilityDate` has to be set in nitro/nuxt config to enable new preset (we could also use two tabs perhaps for now for legacy and new one)

**Comment by gxres042** [comment]
I add a sample `nitro.config.ts` file into it. 🤔 

For `compatibilityDate` option, we should document it at https://nitro.build/config too.

**Review by pi0 — APPROVED** [review]
Thanks! ❤️

**Comment by adamdehaven** [comment]
@pi0 took me a bit of digging when upgrading `nuxt` to `3.14.x` to realize why my app that utilizes the `cloudflare-module` preset was broken 😬 

Turns out, the change to static assets requires this change in my `wrangler.toml`:

```diff
- [site]
- bucket = ".output/public"
+ assets = { directory = "./.output/public/", binding = "ASSETS" }
```

This PR addresses the change in the docs; however, it may be worth a specific callout in the `2.10` release notes

**Comment by pi0** [comment]
Yes, this is required with latest compatibility date being set (otherwise nitro uses legacy `cloudflare_worker` preset that uses old assets handling)

**Pull request merged**